### ISSUE TITLE
Add support for variable price option rates

### ISSUE THREAD

**Issue by mindctrl** [easydigitaldownloads/EDD-Simple-Shipping#31]
I've seen this requested several times now, and I think it would be great to support different rates for each price option, and I think it still keeps the product "simple". Here's a real customer use case:

Photo product with 4 options:
- Lower resolution download
- Higher resolution download
- Print
- Print with frame

Customer wants to charge more shipping for the framed version because of the size and weight differences.

Thoughts?

**Comment by pippinsplugins** [comment]
I agree. Let's make it happen.

**Comment by pippinsplugins** [comment]
@cklosowski Can we put this on your todo list for right after EDD 2.5 is finished?

**Comment by cklosowski** [comment]
I think the logical thing to do here is protect the backwards compatibility. Currently we allow base rates for the product with an opt-in checkbox for each variable price.

In order to not break sites, we need to switch to a 'Domestic' and 'International' rate field for each price ID. If the price ID previously had the checkbox to apply shipping to it, we'll auto fill it with the base rates, otherwise they'll be left blank.

This way sites with variable pricing and shipping on them will be respected.

**Comment by pippinsplugins** [comment]
I agree.

**Comment by cklosowski** [comment]
UX for this so far:
![screen shot 2017-02-02 at 3 57 16 pm](https://cloud.githubusercontent.com/assets/1390124/22572224/4de651d4-e960-11e6-9bad-cb55cfaa899a.png)

**Comment by cklosowski** [comment]
This seems to be working fine. To test backwards compatibility, on `master` setup variable pricing for a download, and set a domestic and international rate. Check the box to ship some of the variable price IDs, leave one or two unchecked.

Then switch to this branch and you should see the base prices set before the update for all the price IDs that had the box checked.

**Comment by pippinsplugins** [comment]
I know this isn't Simple Shipping's fault, but this doesn't work well on small screens:
![screen shot 2017-02-09 at 1 22 56 pm](https://cloud.githubusercontent.com/assets/1034109/22801649/f17af25e-eeca-11e6-86ef-35e3fdc95b1b.png)

@cklosowski @SDavisMedia any interest in taking this as an opportunity to explore how we can make some minor adjustments to the price rows to fix that?

**Comment by pippinsplugins** [comment]
@cklosowski the backwards compat isn't working for me. I have a product that has shipping enabled on prices with a global domestic and international rate entered. I checked out the branch and then added the item to the cart. No shipping was added until I went back and populated the per-price shipping rates.

**Comment by cklosowski** [comment]
Ok, I'll dig into that. I must have missed something in my fallback logic.

**Comment by cklosowski** [comment]
@pippinsplugins ok this fallback is working just fine for me.

I setup master, added a new download with variable pricing, both a domestic and international rate, and checked the box for both price ids to create shipping for them.

I then saved it, checked out `issue/31`, and added it to the cart, all shipping was added correctly.

I'll get some more tests too to be sure.

**Comment by pippinsplugins** [comment]
Ok this is working great now. Not sure what caused the issue before.

Only thing left is to try and make small screens not suck so much (no fault of Simple Shipping so may not be done here).

**Comment by mintplugins** [comment]
This is working for me. 

**What I tested:**
1. On master, I created a new product and set up variable pricing for small, medium, and large. I enabled shipping for all of them and set up domestic and international shipping rates. 

2. Then I changed to the issue/31 branch. All Shipping options for Domestic and International are automatically populated for each price variation. I did not save the download. 

<img width="1007" alt="screen shot 2017-02-13 at 4 14 52 pm" src="https://cloud.githubusercontent.com/assets/1638128/22903839/23ba4d38-f208-11e6-9ff3-54dfeb699cc9.png">

3. I purchased the small version of the product and shipping was correctly applied. 
<img width="496" alt="screen shot 2017-02-13 at 4 19 27 pm" src="https://cloud.githubusercontent.com/assets/1638128/22903856/3c888b9a-f208-11e6-8eb1-1d2b6c36ad49.png">

4. I purchased the medium version of the product and shipping was correctly applied.
<img width="533" alt="screen shot 2017-02-13 at 4 22 33 pm" src="https://cloud.githubusercontent.com/assets/1638128/22903978/ad319134-f208-11e6-950c-4011a8e31951.png">

5. Then I edited the product so that medium had different shipping rates and re-purchased with both a domestic and an international address. Everything came out correct with the correct shipping fees.
